[18.0] [MIG] l10n_it_account_vat_period_end_settlement

odooNextev opened the pull request.

sergiocorato committed:
[ADD] account_vat_period_end_statement - works as is

[FIX] period vat move set equal to date vat period statement

[FIX] code for 7.0

[FIX] other code for 7.0

[FIX] other code for 7.0

[FIX] copyright

eLBati committed:
[FIX] authors and description

[imp] field help

[FIX] in fase di liquidazione iva, quando vado a ricercare i periodi per la liquidazione, questi sono visualizzati solo dall'utente Admin. L'utente "normale" anche se ha diritti di administratore non visualizza i periodi

tommysnello committed:
in account_vat_period_end_statement/report/vat_period_end_statement.py is missing "from openerp.osv import orm"

eLBati committed:
[imp] description

[FIX] liquidazione IVA - impossibile salvare una nuova liquidazione

[REF] Set all the modules of the master branches to "installable": False
moves all modules into a __unported__ directory

tafaRU committed:
Move account_vat_period_end_statement from __unported__ to root folder

Update code using by 'git checkout origin/7.0  -- account_vat_period_end_statement'

Replace OpenERP Italia by Odoo Italia in all header files

Update copyright year for agilebg

Extract the description in README.rst

Improve the manifest file: change version, add OCA as author, remove deprecated key as active

Add OCA icon

Fix 'ImportError: No module named tools.translate'

Fix 'ImportError: No module named decimal_precision'

Fix 'ImportError: No module named report'

Add group tags to show up the missing labels

Commit:
[IMP] move vat_statement_type to l10n_it_account

[FIX] typos

eLBati committed:
[ADD] porting account_vat_period_end_statement_report from https://github.com/alessandrocamilli/test

[IMP] PEP8 and copyright

[IMP] translation

DAMilani committed:
fix calcolo totali iva stampa registri, fix stampa liquidazione controllo tipo conto imposta credit/debit

eLBati committed:
[FIX] report translation

[ADD] missing translatable terms

DAMilani committed:
Fix Travis warning and new translate

fix Travis
<<

fix line too long

Vat registries:
- add last printed page
- tax registries, add type layout
- add flag print only totals

Vat end period statment:
- link header to l10n_it_account
- print payment information

l10n_it_account:
- change header and footer

eLBati committed:
FIX dependency and versions

FIX avoid breaking inheritance of report.minimal_layout

IMP README with images

IMP translations

DAMilani committed:
Add last printed page and interest managment.

fix author description

eLBati committed:
FIX old import
PEP8
README
CamelCase

ADD test VAT statement creation

pedrobaeza committed:
[MIG] Make modules uninstallable

sergiocorato committed:
account vat period end statement set interest flag manually

remove active_id and correct number revision

Fix selection of interest in vat statement

sergiocorato committed:
remove trailing withespaces

OpenCode committed:
Solve problem with tests of account_vat_period_end_statement after add of account_invoice_sequential_dates

Refactoring for variables name to be more clear

sbidoul committed:
[FIX] remove en.po that was erroneously created by transbot

pedrobaeza committed:
[MIG] Rename manifest files

tafaRU committed:
account_vat_period_end_statement: add multicompany rule for account.vat.period.end.statement

account_vat_period_end_statement: change module version and add post-migration script to copy company from periods

eLBati committed:
porting account_vat_period_end_statement to 10
ADD postgresql: "9.2" minimal postgresql version for the daterange method

account_vat_period_end_statement ADD description on statement.generic.account.line

account_vat_period_end_statement ADD crediti d'imposta
ADD interessi dovuti e modifica del loro calcolo

FIX
ValueError: Expected singleton: account.vat.period.end.statement(1, 2, 4, 6, 7, 8, 9, 10, 12, 13, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27)

tafaRU committed:
[ADD] account_vat_period_end_statement: add migration script for copying account.period (with vat statement) to date.range (#408)

Commit:
[FIX][10]: correzione calcolo liquidazione iva in caso di fatture con integrazione iva (#379)

fix imponibile imposta figlia con integrazione iva

tafaRU committed:
[FIX] account_vat_period_end_statement: check account_peiod table existance in the migration script (#411)

Commit:
modificate traduzioni nella stampa della liquidazione (#413)

tafaRU committed:
[FIX] account_vat_period_end_statement: get migration script multi company compliant (#428)

[FIX] account_vat_period_end_statement: check move_id exists to get statement_draft properly working (#435)

move_id needs to be checked if exists in the computed fields, otherwise the method unlink, during the fields recomputation, raises a MissingError for them

[FIX] account_vat_period_end_statement: post move + cancel move (#436)

[IMP] account_vat_period_end_statement: allow to adviser to manage date.range and date.range.type
Otherwise a user who belongs to adviser group cannot add a date.range to vat statement

[FIX] Move menu related to date range to l10n_it_account module

eLBati committed:
FIX account_vat_period_end_statement tests (#459)

In cases like today, feb 28, no date range is found

SimoRubi committed:
[FIX]  account_vat_period_end_statement README description (#477)

[FIX] README format issues (#501)

jackjack82 committed:
[11][MIG] Migrating account vat statement to 11

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 99.1% (109 of 110 strings)

Translation: l10n-italy-10.0/l10n-italy-10.0-account_vat_period_end_statement
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-10-0/l10n-italy-10-0-account_vat_period_end_statement/it/

SimoRubi committed:
[MIG] account_vat_period_end_statement: Migration to 12.0

eLBati committed:
IMP account_vat_period_end_statement views (#4)

SimoRubi committed:
[FIX] account_vat_period_end_statement: New statements had the reconciled check set to True, but it should be False

eLBati committed:
IMP account_vat_period_end_statement report: do not print rows with zero values

eLBati committed:
IMP account_vat_period_end_statement preventing to manually add lines to debit and credit VAT
because no use case is expected to allow it and they are removed when user clicks on "recompute"

eLBati committed:
IMP account_vat_period_end_statement: set previous year credit

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 98.2% (110 of 112 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-account_vat_period_end_statement
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-account_vat_period_end_statement/it/

jado95 committed:
12.0-fix-report-account_vat_period_end_statement

eLBati committed:
Translated using Weblate (Italian)

Currently translated at 99.1% (112 of 113 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-account_vat_period_end_statement
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-account_vat_period_end_statement/it/

eLBati committed:
IMP account_vat_period_end_statement adding down payment  handling

eLBati committed:
Translated using Weblate (Italian)

Currently translated at 99.1% (115 of 116 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-account_vat_period_end_statement
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-account_vat_period_end_statement/it/

eLBati committed:
IMP account_vat_period_end_statement allowing to create an annual prospect, without accounting entries

eLBati committed:
Annual prospect of VAT statement: do not compute previous credits.
Add annual field to list and search view

eLBati committed:
Translated using Weblate (Italian)

Currently translated at 99.2% (118 of 119 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-account_vat_period_end_statement
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-account_vat_period_end_statement/it/

Commit:
merge PR 1264 1717 1728 1737 1739

eLBati committed:
FIX account_vat_period_end_statement: do not consider annual statements while computing previous credits in regular statements

eLBati committed:
FIX account_vat_period_end_statement: while computing "previous_year_credit" consider periods (date ranges) instead of statement date, because typically end year statement is done in successive year

robyf70 committed:
[12.0][account_vat_period_end_statement] Add more fields such as previous credit/debit and interests and tax credit on the tree view

fredzamoabg committed:
[IMP] account_vat_period_end_statement: black, isort, prettier

fredzamoabg committed:
[MIG] account_vat_period_end_statement: Migration to 14.0

account_vat_period_end_statement: Migration to 14.0 - FIX tests, views and readme - eLBati

eLBati committed:
account_vat_period_end_statement: print down payment and computation method

eLBati committed:
Translated using Weblate (Italian)

Currently translated at 95.2% (119 of 125 strings)

Translation: l10n-italy-12.0/l10n-italy-12.0-account_vat_period_end_statement
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-12-0/l10n-italy-12-0-account_vat_period_end_statement/it/

jado95 committed:
[IMP] Add new field to filter the taxes to be processed, so the account configured in the taxes "Account used for VAT settlement" must be indicated.

SimoRubi committed:
[FIX] account_vat_period_end_statement: Add default repartition lines in taxes, use default company journals
Also use common setup for invoicing tests

TheMule71 committed:
[FIX] account_vat_period_end_statement: fix test tags

TheMule71 committed:
[FIX] account_vat_period_end_statement: Select previous Statement by Account

Co-Authored-by: SirTakobi <sir@takobi.online>

michelerusti committed:
[MIG] account_vat_period_end_statement: Migration to 16.0

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 94.2% (114 of 121 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-account_vat_period_end_statement
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-account_vat_period_end_statement/it/

Borruso committed:
[FIX] account_vat_period_end_statement: fix align amounts

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 67.7% (82 of 121 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-account_vat_period_end_statement
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-account_vat_period_end_statement/sl/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (121 of 121 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-account_vat_period_end_statement
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-account_vat_period_end_statement/it/

michelerusti committed:
[16.0][FIX] - account_vat_period_end_statement - wrong form proportions

michelerusti committed:
[16.0][FIX] Remove unused css classes

Borruso committed:
[FIX] account_vat_period_end_statement: fix nolabel view

SirAionTech committed:
Updated from template

SirAionTech committed:
[REF] account_vat_period_end_statement: Update repo template

SirAionTech committed:
[REF] account_vat_period_end_statement: Reformatted after template update

SirAionTech committed:
[FIX] account_vat_period_end_statement: Create move with payment term
Method account.payment.term.compute does not exist

sergiocorato committed:
[16.0][FIX] account_vat_period_end_statement order on account.vat.period.end.statement

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (122 of 122 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-account_vat_period_end_statement
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-account_vat_period_end_statement/it/

SirAionTech committed:
Updated from template

SirAionTech committed:
[COV] account_vat_period_end_statement: Avoid leap year
Otherwise on February 29th the following happens:
>>> date(today.year - 1, today.month, today.day)
ValueError: day is out of range for month

SirAionTech committed:
[REF] account_vat_period_end_statement: Restore contributor

Accidentally removed in https://github.com/OCA/l10n-italy/commit/dc9646351a69fd53e14a6b11e784f6adfeb3482a

SirAionTech committed:
[REF] account_vat_period_end_statement: Allow to re-use test methods

salvorapi committed:
[FIX] account_vat_period_end_statement: remove not required dependency

Commit:
[IMP] account_vat_period_end_statement: Updated "Total Statement" label

Added (credit) or (debit) to "Total Statement" label based on the value of the total amount

Co-authored-by: odooNextev <odoo@nextev.it>

SirAionTech committed:
[FIX] account_vat_period_end_statement: Repeated payment info

SirAionTech committed:
[REF] account_vat_period_end_statement: Allow to reuse to pay table

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (122 of 122 strings)

Translation: l10n-italy-16.0/l10n-italy-16.0-account_vat_period_end_statement
Translate-URL: https://translation.odoo-community.org/projects/l10n-italy-16-0/l10n-italy-16-0-account_vat_period_end_statement/it/

primes2h committed:
[FIX] account_vat_period_end_statement: untranslatable strings

odooNextev committed:
[IMP] account_vat_period_end_statement: pre-commit auto fixes

primes2h commented:
/ocabot migration account_vat_period_end_statement

odooNextev committed:
[MIG] account_vat_period_end_statement: Migration to 18.0

primes2h reviewed (changes requested):
https://github.com/OCA/l10n-italy/issues/4510

odooNextev commented:
> #4510

Rinomino il modulo in `account_vat_period_end_settlement` oltre ai riferimenti nel codice?

primes2h commented:
> > #4510
> 
> Rinomino il modulo in `account_vat_period_end_settlement` oltre ai riferimenti nel codice?

In questo caso potremmo anche approfittare dell'occasione per uniformare e semplificare il nome.

`l10n_it_vat_settlement`

@OCA/local-italy-developers  che ne pensate?

eLBati commented:
@odooNextev questa PR sostituisce questa https://github.com/OCA/l10n-italy/pull/4565 ?

odooNextev commented:
> @odooNextev questa PR sostituisce questa #4565 ?

Sì

eLBati commented:
> > @odooNextev questa PR sostituisce questa #4565 ?
> 
> Sì

Nel senso che la https://github.com/OCA/l10n-italy/pull/4565 era incompleta?

odooNextev commented:
> > > @odooNextev questa PR sostituisce questa #4565 ?
> > 
> > 
> > Sì
> 
> Nel senso che la #4565 era incompleta?

A dire il vero la #4565 non era segnalata nella issue base di migrazione e non l'avevo trovata, così ne ho cominciata una nuova

odooNextev committed:
[MIG] account_vat_period_end_statement: rename to account_vat_period_end_settlement

eLBati committed:
account_vat_period_end_statement: REF removing cee_type

francesco-ooops commented:
includere https://github.com/OCA/l10n-italy/pull/4204

micheledic commented on l10n_it_account_vat_period_end_settlement/__manifest__.py:
```diff
@@ -0,0 +1,47 @@
+    "assets": {
+        "web.assets_backend": [
+            "l10n_it_account_vat_period_end_settlement/static/src/css/"
+            "l10n_it_account_vat_period_end_settlement.css",
```
Modificare o qui il nome file o il nome file nella folder static/src/css che attualmente è account_vat_period_end_statement

micheledic replied:
Cambiare in static/css il css in 
`l10n_it_account_vat_period_end_settlement.css`

micheledic commented on l10n_it_account_vat_period_end_settlement/report/vat_statement.py:
```diff
@@ -0,0 +1,85 @@
+
+
+class VatPeriodEndStatementReport(models.AbstractModel):
+    _name = "report.account_vat_period_end_statement.vat_statement"
```
Rinominare in 
`_name = "report.l10n_it_account_vat_period_end_settlement.vat_statement"`
altrimenti non funziona

Borruso commented on l10n_it_account_vat_period_end_settlement/static/src/css/account_vat_period_end_statement.css:
potresti cambiare il nome del file con `l10n_it_account_vat_period_end_settlement.css`

MaurizioConte commented:
Ciao,
ho provato a fare i test su "runboat" ma su entrambi i database ci sono problemi, sul primo DB in elenco mi viene presentato una videata che tutto è fuorché Odoo mentre sul secondo DB accedo ma quando calcolo la liquidazione esce vuota.

Ho provato a testarlo sul mio DB che è già alla 18.0, migrato dalla 16, e c'è un errore in quanto viene esposta solo l'imposta di acquisto con l'imponibile e senza valore dell'iva mentre per la parte di vendita l'imposta non figura proprio.
Essendo l'imposta unica mentre i conti contabili iva utilizzati sono 2 non Riesco a capire come faremo a considerare in liquidazione iva la metodologia di integrazione iva.
Anche perché i dati delle operazioni intra ed extra vanno comunicati all'ADE in LIPE - Comunicazione Dati IVA.

Attendo un vostro riscontro magari anche per confrontarci.
Saluti

francesco-ooops commented:
In effetti il runboat in questa PR è spaccato

`The style compilation failed. This is an administrator or developer error that must be fixed for the entire database before continuing working. See browser console or server logs for details.
`

micheledic commented:
> In effetti il runboat in questa PR è spaccato
> 
> `The style compilation failed. This is an administrator or developer error that must be fixed for the entire database before continuing working. See browser console or server logs for details. `

E' il nome del css errato, sopra ho fatto review ;)

eLBati committed:
[REF] l10n_it_account
- Adding _get_credit_accounts and _get_debit_accounts
- Compute deductible_balance in case of split payment
and comments
- Add customer_balance and supplier_balance for VAT documents

eLBati committed:
[IMP] l10n_it_account_vat_period_end_settlement: statement > settlement

eLBati committed:
[FIX] l10n_it_account_vat_period_end_settlement missing CSS

eLBati committed:
[MIG] REF l10n_it_account_vat_period_end_settlement
- Remove vat_statement_account_id
- Adapt to reverse charge cases
- Adapt to split payment cases
- Fix reconciliation process
- REF tests

eLBati committed:
[FIX] l10n_it_vat_registries to handle split payment case

VBNext reviewed (approved):
Funzionale OK!!!!

stenext reviewed (approved):
LGTM

eLBati commented:
Metto patch visto che ci sono anche correzioni ad altri moduli

/ocabot merge patch
